### PR TITLE
Model refactor #1

### PR DESCRIPTION
**Pull request by IstvanM**
Refactored:
- Project
- EventDataTable
- EvendDef
- EventFieldDef

TLDR: The main problem I am solving here is that during pickling the references are stored as well.. This makes the stored documents very big.

Event Data Table had a Project type property. Same for EventDef and EventFieldDef. This is a problem because during the Storage save of Dashboard or SavedMetric. The project was embedded to the stored objects. If the user changed the project on the website later it didn't update the already saved Dashboards and Saved Metrics. 
So I have introduced the State Class (this is by class definition is not pickled as the `def __get_state__()` returns None.
However we need to set the State by hand. With this I am kind of writing a new ORM framework which is not good. However I would like to avoid using SQL Alchemy for this, because the best DB for Mitzu would by Dynamo or Mongo. SQL Alchemy doesn't really support those....

I have added many bugfixes to the PR as well.
One thing is still missing in the PR ( but I will add that later in another PR ). The Project class has a Connection reference that is not a State. So every saved project in the Storage will embed the Connection object which again causes the same issue.

### CODE REVIEW

File: mitzu/__init__.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 ]
 
 
-def load_from_project_file(
-    project: str, folder: str = "./", extension="mitzu"
-) -> DiscoveredProject:
-    return DiscoveredProject.load_from_project_file(project, folder, extension)
+# def load_from_project_file(
```
Comment (IstvanM) [inline mitzu/__init__.py:34]: I will fix this, I need to do some small changes around this.

File: mitzu/adapters/sqlalchemy_adapter.py [hunk 2]
```diff
@@ -513,7 +513,6 @@ def get_field_enums(
                     f: M.EventFieldDef(
                         _event_name=evt,
                         _field=f,
-                        _project=self.project,
```
Comment (IstvanM) [inline mitzu/adapters/sqlalchemy_adapter.py:516]: Here I am not setting anymore the project. The EventDataTable has the Project as State.

File: mitzu/adapters/sqlite_adapter.py [hunk 3]
```diff
@@ -22,7 +22,10 @@ def _column_index_support(self):
         return False
 
     def _get_connection_url(self, con: M.Connection):
-        return "sqlite://?check_same_thread=False"
+        if not con.host:
+            return "sqlite://?check_same_thread=False"
```
Comment (IstvanM) [inline mitzu/adapters/sqlite_adapter.py:26]: This allows SQLite connections in the Webapp finally. This is good because with `docker run mitzuio/mitzu` the user already has an example DWH.

File: mitzu/helper.py [hunk 4]
```diff
@@ -53,3 +54,7 @@ def get_segment_project(segment: M.Segment) -> M.Project:
         return get_segment_project(segment._left)
     else:
         raise ValueError(f"Segment is of invalid type: {type(segment)}")
+
+
+def create_unique_id():
```
Comment (IstvanM) [inline mitzu/helper.py:59]: This piece of code was all over the place. So I put it to the helper.

File: mitzu/model.py [hunk 5]
```diff
@@ -436,7 +433,7 @@ class Connection:
     url_params: Optional[str] = None
     # Used for adapter configuration
     extra_configs: Dict[str, Any] = field(default_factory=dict)
-    _secret: ProtectedState[str] = field(default_factory=ProtectedState)
+    _secret: State[str] = field(default_factory=State)
```
Comment (IstvanM) [inline mitzu/model.py:436]: ProtectedState is now Called State. (The original State was still picklable)

File: mitzu/model.py [hunk 6]
```diff
@@ -896,55 +883,20 @@ def _get_path(
         else:
             return pathlib.Path(folder, f"{project_name}.{extension}")
 
-    def serialize(self) -> str:
```
Comment (IstvanM) [inline mitzu/model.py:899]: I moved this to the Serialization module.

File: mitzu/model.py [hunk 7]
```diff
@@ -1004,28 +956,26 @@ def _get_name(self) -> str:
...
     _fields: Dict[Field, EventFieldDef]
-
-    # TBD remove project
-    _project: Project
-
-    # TBD remove EDT
     _event_data_table: EventDataTable
     _description: Optional[str] = ""
 
+    @property
```
Comment (IstvanM) [inline mitzu/model.py:975]: Python property is a method that you can refer to as a property... but it is essentially just a method.

File: mitzu/webapp/storage.py [hunk 8]
```diff
@@ -29,42 +29,54 @@
 SIMPLE_CHART_PREFIX = "__simple_chart__"
 SAVED_METRIC_PREFIX = "__saved_metric__"
 DASHBOARD_PREFIX = "__dashboard__"
+SAMPLE_PROJECT_ID = "sample_project_id"
 
 
-def create_sample_project() -> M.DiscoveredProject:
+def setup_sample_project(storage: MitzuStorage):
```
Comment (IstvanM) [inline mitzu/webapp/storage.py:35]: Here I add optionally the Sample SQLite DWH to the Storage.

File: tests/unit/webapp/pages/test_explore_page.py [hunk 9]
```diff
@@ -58,14 +57,45 @@ def find_component_by_id(
 
 @fixture(scope="module")
 def discovered_project():
-    path = os.path.dirname(os.path.abspath(__file__))
```
Comment (IstvanM) [inline tests/unit/webapp/pages/test_explore_page.py:61]: This made this test fail everytime we changed the Model classes. Now it is more stable.

**Review by medgyes — DISMISSED** [review]
